### PR TITLE
Use .Open queues in public PRs, non-open in internal

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes https://github.com/dotnet/runtime/issues/65741

I had to update some queues based on the [platforms supported for 3.1](https://github.com/dotnet/core/blob/main/release-notes/3.1/3.1-supported-os.md):
  - Windows 19H1 to 20H2
  - MacOS 1013 and 1014 to 1015 and 1100

Made sure all the queue names matched something that existed here: https://helix.dot.net/

The sample run is here: https://dev.azure.com/dnceng/internal/_build/results?buildId=1641884&view=results

Kudos to @hoyosjs for helping me find the right places to change.

### CODE REVIEW

**Comment by carlossanlop** [comment]
My knowledge about this is very limited. I assume it's expected to see the CI failing, here and in the internal run.

File: eng/pipelines/linux.yml
Comment (MattGal) [inline eng/pipelines/linux.yml]: While docker images live more or less forever (as long as their registry doesn't purge them), Alpine 3.9 has been EOL since Nov 2020.

File: eng/pipelines/linux.yml
Comment (MattGal) [inline eng/pipelines/linux.yml]: Same comments as before, Alpine 3.8 was end of life in May 2020.

File: eng/pipelines/linux.yml
Comment (carlossanlop) [reply eng/pipelines/linux.yml]: Can you please tell me what name to use instead?

Comment (carlossanlop) [reply eng/pipelines/linux.yml]: I ask because I don't see that string in https://helix.dot.net/ , so I don't know how that's decided.

Comment (MattGal) [reply eng/pipelines/linux.yml]: You can get tags from https://raw.githubusercontent.com/dotnet/versions/main/build-info/docker/image-info.dotnet-dotnet-buildtools-prereqs-docker-main.json .  For instance you might try `mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.12-helix-20211214164128-aca80d8`

**Comment by MattGal** [comment]
> My knowledge about this is very limited. I assume it's expected to see the CI failing, here and in the internal run.

If you expect and ignore CI failures you should probably just remove testing since testing costs time and money and you're saying it's normal to be ignoring the outcome of such testing.

The errors are actually interesting and should likely be investigated.  Specifically it's complaining about not finding https://dnceng.visualstudio.com/public/_artifacts/feed/dotnet-eng/NuGet/Microsoft.DotNet.Helix.Sdk/2.0.0-beta.21552.6/overview despite https://github.com/dotnet/corefx/blob/release/3.1/NuGet.config#L20 containing a reference to this feed.  I'd guess something might be overriding which nuget.config file is used. 


**EDIT**: This is an AzDO outage, from the logs:

``` 
The HTTP request to 'GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/microsoft.dotnet.helix.sdk/index.json' has timed out after 100000ms.
Retrying 'FindPackagesByIdAsync' for source 'https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/d1622942-d16f-48e5-bc83-96f4539e7601/nuget/v3/flat2/microsoft.dotnet.helix.sdk/index.json'.
Response status code does not indicate success: 500 (Internal Server Error - Maximum waiter thread limit 500 is reached. Throwing exception to avoid thread exhaustion. (DevOps Activity ID: 5AD5BDCF-9E23-45A3-BD21-13199A4EAFF8)).
```
... so retry should work.

**Comment by carlossanlop** [comment]
> If you expect and ignore CI failures

I did not say we should ignore the failures. I said seeing failures might be expected, based on how complicated the servicing process for 3.1 has been and there are many issues piled one on top of another.

> you should probably just remove testing since testing costs time and money

Not sure why that is being suggested. Is that something that is being planned? Regardless, it's outside the scope of this PR that I was asked to work on.

> and you're saying it's normal to be ignoring the outcome of such testing.

Once again, I never said ignore.

File: eng/pipelines/linux.yml
Comment (hoyosjs) [reply eng/pipelines/linux.yml]: SGTM. Core 3.1 supports 3.12+ https://github.com/dotnet/core/blob/main/release-notes/3.1/3.1-supported-os.md

**Comment by hoyosjs** [comment]
> If you expect and ignore CI failures you should probably just remove testing since testing costs time and money and you're saying it's normal to be ignoring the outcome of such testing.

Nope, this is for internal PRs (MSRC testing). 

Sadly, the internal run didn't test the codepath being tested. For this to run we need `eq(${{ parameters.isOfficialBuild }}, False)` and we have `${{ and(ne(variables['System.TeamProject'], 'public'), notIn(variables['Build.Reason'], 'PullRequest')) }}`. For manual builds, we get:

```
Evaluating: and(ne(variables['System.TeamProject'], 'public'), notIn(variables['Build.Reason'], 'PullRequest'))
Expanded: and(ne('internal', 'public'), notIn('Manual', 'PullRequest'))
Result: True
```

So, for this to actually use those queues we need `isOfficialBuild` = False. The only way to achieve that is to create a PR.  This is also going to hit one other snag. `HelixApiAccessToken` needs to be set, and we set it only for official builds here https://github.com/dotnet/corefx/blob/97f1387470b692af428e5c2c4893cecc0fc08dea/eng/pipelines/corefx-base.yml#L221-L223

I would relax this condition to be if running on internal, make the helix token available, and we need to remove creator in the next line on internal (instead of isOfficialBuild only set it on public)

**Comment by hoyosjs** [comment]
Sample build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1658409&view=results

@mmitche Given this is to enable MSRC PR testing, what's the timeline for this one?

@ericstj @jeffhandley any idea who could help with a review here? There's some changes that will probably need scrutiny (Several of the old queues were dead, and this can have testing fallout)

**Comment by hoyosjs** [comment]
For macOS:
- X509 tests fail a bunch of tests, mostly around expiration verification and chain verification. At least some of them would go away with https://github.com/dotnet/corefx/pull/43132

For Linux-musl: this one is scary - a lot of Http test failures. 

For Windows:
- All are from EventLogTraceListener. This had happened in the past in https://github.com/dotnet/runtime/issues/30565 where retry was failing.
- Some System.Io failures. These one seem legitimate, but can't tell.

*nix in general:
- Some Globalization IDNA conformance test failed in macOS 11.00 and Alpine. They probably use a newer version of Unicode https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-corefx-refs-pull-43135-merge-1b06bc9361344aa19e/System.Globalization.Extensions.Tests/1/console.806b00c1.log?sv=2019-07-07&se=2022-03-31T11%3A33%3A39Z&sr=c&sp=rl&sig=CS9s6NRnA4wQva%2BdE3lXxKCEaprF2nesy3NJrmvmtos%3D This was fixed in RT, see https://github.com/dotnet/runtime/issues/29224. @tarekgh does this need to be ported?

File: eng/pipelines/linux.yml
Comment (bartonjs) [inline eng/pipelines/linux.yml:122]: Instead of an opposing condition if, would this be better as an else?

File: eng/pipelines/windows.yml
Comment (bartonjs) [inline eng/pipelines/windows.yml:119]: In the linux files the parens were escaped with backslashes, here it's backquote.  Is that expected?

File: eng/pipelines/linux.yml
Comment (bartonjs) [inline eng/pipelines/linux.yml:139]: I went through and checked .Open and not have parity, and that before and after have parity (aside from replacing Alpine 3.9 with Alpine 3.14).  No one else has to do that multi-diff if they don't want to.

File: eng/pipelines/windows.yml
Comment (bartonjs) [inline eng/pipelines/windows.yml:124]: Before had an ES build, which I assume means the OS is in Spanish.  After does not.

File: eng/pipelines/windows.yml
Comment (bartonjs) [inline eng/pipelines/windows.yml:119]: I like that the nanoQueues definition is back; but to actually restore Nano we need to undo the two other edits from

https://github.com/dotnet/corefx/pull/43136/commits/8ebe7bbe421fa2fbcfcf754cfbe5979fffa2863c

(Lines 38 and 64)

File: eng/pipelines/linux.yml
Comment (hoyosjs) [reply eng/pipelines/linux.yml:122]: Was trying to change as little as possible from the original yml. Back then else didn't exist.

File: eng/pipelines/windows.yml
Comment (hoyosjs) [reply eng/pipelines/windows.yml:119]: This was there before and a merge from 3.1 removed it - not sure why they were inconsistent. Both are valid escapes afaik.

File: eng/pipelines/linux.yml
Comment (bartonjs) [reply eng/pipelines/linux.yml:122]: Fair, fair :)

**Comment by bartonjs** [comment]
> Some System.Io failures. These one seem legitimate, but can't tell.

I don't see them now (maybe I looked in the wrong place).  If they were all `System.IO.Ports.Tests.OpenDevices.OpenDevices01` failing with "unknown error 5", that's a known issue that needs a test-only backport to 3.1 (with a backport issue opened somewhere).

File: eng/pipelines/windows.yml
Comment (hoyosjs) [reply eng/pipelines/windows.yml:124]: There's no client queue with ES anymore

Comment (bartonjs) [reply eng/pipelines/windows.yml:124]: That's... unfortunate.  But, if that's just where things are, then I guess we have to take what we can get.

**Comment by bartonjs** [comment]
If we can get the two lines added back to restore the Nano testing, LGTM.

**Review by bartonjs — APPROVED** [review]

**Comment by hoyosjs** [comment]
@bartonjs they are in the internal build I added - and any opinion of Http and Io failures? They are not the one you pointed out. For example https://dev.azure.com/dnceng/internal/_build/results?buildId=1658409&view=ms.vss-test-web.build-test-results-tab&runId=45680330&resultId=154547&paneView=debug

**Comment by bartonjs** [comment]
Ah. Those IO failures look like https://github.com/dotnet/runtime/issues/20986, which means we probably need to backport the fix (@dotnet/area-system-io).  Not sure why it isn't failing in the .Open versions.

**Comment by bartonjs** [comment]
@wfurt Can you look at the HTTP failures in https://dev.azure.com/dnceng/internal/_build/results?buildId=1658409&view=ms.vss-test-web.build-test-results-tab&runId=45680330&resultId=154547&paneView=debug and say if any of them are reasons that we should hold off on accepting these helix queue changes?  I didn't see similar issues/failures in dotnet/runtime.

**Comment by bartonjs** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=1658409&view=ms.vss-test-web.build-test-results-tab is probably the more useful link for the HTTP test failures.  (GetAsyncVersion11_BadResponseVersion_ThrowsOr00 and maybe some  friends)

**Comment by tarekgh** [comment]
@hoyosjs 

> Some Globalization IDNA conformance test failed in macOS 11.00 and Alpine. They probably use a newer version of Unicode [helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-corefx-refs-pull-43135-merge-1b06bc9361344aa19e/System.Globalization.Extensions.Tests/1/console.806b00c1.log?sv=2019-07-07&se=2022-03-31T11%3A33%3A39Z&sr=c&sp=rl&sig=CS9s6NRnA4wQva%2BdE3lXxKCEaprF2nesy3NJrmvmtos%3D](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-corefx-refs-pull-43135-merge-1b06bc9361344aa19e/System.Globalization.Extensions.Tests/1/console.806b00c1.log?sv=2019-07-07&se=2022-03-31T11%3A33%3A39Z&sr=c&sp=rl&sig=CS9s6NRnA4wQva%2BdE3lXxKCEaprF2nesy3NJrmvmtos%3D) This was fixed in RT, see https://github.com/dotnet/runtime/issues/29224. @tarekgh Tarek Mahmoud Sayed FTE does this need to be ported?

My Understanding is Alpine enable the Globalization Invariant Mode which I expect the IDN will not work perfectly. I agree this is most properly because the Unicode test file version used. If porting the fix here will help, please try it. I can help looking if this is didn't help.

**Comment by wfurt** [comment]
he test failures looks weird. I don't remember seeing anything like this. Do we know if everything looks green on other PRs or the release build? This was the "PlatformHandler" e.g. when we integrate with libcurl. We dump that I think around 5.0 so we did not touch it for very long time.

On. side done, adding new queues should not impact Linux runs. So what ever that I feel it is not caused by this PR.

**Comment by bartonjs** [comment]
Sounds to me like we can merge this as-is, and clean up any other test debt on the other side of the merge.

**Pull request merged**
